src/replayer.py

```diff
@@ -0,0 +1,306 @@
+
+    _args = parser.parse_args()
+
+    # Override arguments with environment variables where set
+    if "AWS_PROFILE" in os.environ:
+        _args.aws_profile = os.environ["AWS_PROFILE"]
+
+    if "AWS_REGION" in os.environ:
+        _args.aws_region = os.environ["AWS_REGION"]
+
+    if "ENVIRONMENT" in os.environ:
+        _args.environment = os.environ["ENVIRONMENT"]
+
+    if "APPLICATION" in os.environ:
+        _args.application = os.environ["APPLICATION"]
+
+    if "LOG_LEVEL" in os.environ:
+        _args.application = os.environ["LOG_LEVEL"]
+
+    if "API_HOSTNAME" in os.environ:
+        _args.application = os.environ["API_HOSTNAME"]
+
+    required_args = ["API_HOSTNAME"]
+    missing_args = []
+    for required_message_key in required_args:
+        if required_message_key not in _args:
+            missing_args.append(required_message_key)
+    if missing_args:
+        raise argparse.ArgumentError(
+            None,
+            "ArgumentError: The following required arguments are missing: {}".format(
+                ", ".join(missing_args)
+            ),
+        )
+
+    return _args
+
+
+args = get_parameters()
+logger = setup_logging(args.log_level)
+region = os.environ.get("AWS_REGION")
+
+
+def handler(event, context):
+
+    session = boto3.session.Session()
+    default_credentials = session.get_credentials().get_frozen_credentials()
+
+    loaded_event = json.loads(event)
+    try:
+        original_request = loaded_event["Body"]["originalRequest"]
+        original_response = loaded_event["Body"]["originalResponse"]
+    except Exception as e:
+        logger.error("Attempted to extract event items but was unable.")
+
+    nino = original_request.get("nino")
+    transaction_id = original_request.get("transaction_id")
+    from_date = original_request.get("from_date")
+    to_date = original_request.get("to_date")
+
+    actual_response = replay_original_request(
+        default_credentials, nino, transaction_id, from_date, to_date
+    )
+
+    decrypted_original_response = decrypt_response(original_response, original_request)
+    decrypted_actual_response = decrypt_response(actual_response, original_request)
+
+    if compare_responses(
+        decrypted_original_response, decrypted_actual_response, original_request
+    ):
+        logger.info('Final result", "status": "match')
+    else:
+        logger.info('Final result", "status": "miss')
+
+
+def replay_original_request(
+    default_credentials, nino, transaction_id, from_date, to_date
+):
+    auth = AWSRequestsAuth(
+        aws_access_key=default_credentials.access_key,
+        aws_secret_access_key=default_credentials.secret_key,
+        aws_token=default_credentials.token,
+        aws_host=f"{args.api_hostname}",
+        aws_region=f"{args.aws_region}",
+        aws_service="execute-api",
+    )
+
+    request_parameters = f"nino={nino}&transactionId={transaction_id}&fromDate={from_date}&toDo={to_date}"
```

Comment:
typo
```suggestion
    request_parameters = f"nino={nino}&transactionId={transaction_id}&fromDate={from_date}&toDate={to_date}"
```
also needs to be able to query without dates